rest-parameters.md

```diff
@@ -0,0 +1,64 @@
+# ES6 Rest parameters
+
+Functions can take a rest parameters that act as a placeholder for a non-finite number of arguments.
+
+```javascript
+function score(user, ...marks) {
+  const marks_count = marks.length;
+  let total = 0;
+
+  // sum all the marks together
+  for (let mark of marks) {
+    total += mark;
+  }
+
+  const score = Math.round(total / marks_count);
+
+  return `John has a score of ${score}`;
+}
+
+console.log(score("John", 12, 8, 8, 12, 15));
+
+// "John has a score of 11"
+```
+
+Rest parameters automatically build an array, so given this function:
```

Comment:
It's an array `instance`, if we highlight that with something like: ```Rest parameters give an array instance. `pop`, `push`, `join` and other array methods are available.```

```js
function sentence(...words) {
  return words.join(" ");
}
console.log(sentence("You", "are", "awesome")); // "You are awesome"
```
This way we are keeping the code and execution output together.

Then you could mention that `for/of` is also available and have an `avg` function with the first code example.